app/src/main/java/org/mozilla/fenix/components/metrics/MetricController.kt

```diff
@@ -267,6 +268,29 @@ internal class ReleaseMetricController(
             }
             null
         }
+
+        Component.LIB_DATAPROTECT to SecurePrefsReliabilityExperiment.Companion.Actions -> {
+            when (item) {
+                // Map each type of experiment with a result.
```

Comment:
You don't need to do anything with enums. Just record `value` directly into an extra, and make sure to tease out exception classes from the Fact's metadata and pass these into corresponding extras, as well.